### PR TITLE
COMMERCE-9727 API requests for product's subscription configuration are not working

### PR DESCRIPTION
**Pull request by ghost**
https://issues.liferay.com/browse/COMMERCE-9727

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSubscriptionConfigurationUtil.java [hunk 1]
```diff
@@ -49,7 +50,11 @@ public static CPDefinition updateCPDefinitionSubscriptionInfo(
 			GetterUtil.get(
 				productSubscriptionConfiguration.getLength(),
 				cpDefinition.getSubscriptionLength()),
-			subscriptionTypeValue, null,
+			subscriptionTypeValue,
+			UnicodePropertiesBuilder.create(
+				productSubscriptionConfiguration.getSubscriptionTypeSettings(),
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSubscriptionConfigurationUtil.java:55]: what if this is null?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
@CrescenzoRegaLF can you please check my comment? Thanks!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSubscriptionConfigurationUtil.java (hunk 1 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSubscriptionConfigurationUtil.java:55]: the method UnicodePropertiesBuilder.create check that the map is not null, if the map is null it return a Collections. <String, String> emptyMap()

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSubscriptionConfigurationUtil.java:55]: thank you!

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:forward

**Pull request closed without merging**